### PR TITLE
feat(types): introduce `Timestamptz` type

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR introduces `Timestamptz` type and array. It's another attempt after #7084.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] ~~I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [ ] My PR contains user-facing changes.

### CODE REVIEW

**Comment by BugenZhao** [comment]
May I ask whether we're going to unify logical types (`DataType`) and physical types (`ScalarImpl::..`) after this PR?

**Comment by wangrunji0408** [comment]
> May I ask whether we're going to unify logical types (`DataType`) and physical types (`ScalarImpl::..`) after this PR?
> 
> Seems still impossible since there could some metadata in `DataType` like the field name of structs. But we can now establish 1v1 relationship **from** logical types **to** physical types.

Exactly. Even though we have a 1-1 mapping between the variants of `DataType` and `ScalarImpl` (which is good), for some types it still requires the metadata in `DataType` to interpret values in `ScalarImpl`.

**Comment by xiangjinwu** [comment]
One example is `ScalarImpl::List` with a `ListValue` containing `[]` or `[None]`. It is still necessary to have `DataType` to known whether it is `int[]` or `varchar[]` or `int[][]`.

**Comment by xiangjinwu** [comment]
Oh sorry the build failure is due to an uncaught conflict with #10462. Should be an easy fix. I am reviewing the other parts

File: src/stream/src/executor/aggregation/agg_impl/mod.rs [hunk 1]
```diff
@@ -215,7 +215,12 @@ pub fn create_streaming_agg_impl(
                     (Min, time, time, StreamingMinAgg::<TimeArray>),
                     (Min, date, date, StreamingMinAgg::<DateArray>),
                     (Min, timestamp, timestamp, StreamingMinAgg::<TimestampArray>),
-                    (Min, timestamptz, timestamptz, StreamingMinAgg::<I64Array>),
+                    (
+                        Min,
```
Comment (xiangjinwu) [inline src/stream/src/executor/aggregation/agg_impl/mod.rs:219]: Missing `(Count, timestamptz, int64, StreamingCountAgg::<TimestamptzArray>),`

**Review by xiangjinwu — APPROVED** [review]
Rest lgtm. Thank you!

**Pull request merged**